### DISCUSSION TITLE
[교재 59, 61 page] 1.4.2 App 컴포넌트 내용중 App.js 가 아니라 App.tsx가 만들어졌는데요?

### DISCUSSION THREAD

**Discussion by digerati73** [velopert/dealing-with-react-native#48] [Q&A]
1. 교재 59  1.4.2 App 컴포넌트 내용중 App.js 가 아니라 App.tsx가 만들어졌는데요

상관없나요?

물론 58페이지 까지 해서 실물 아이폰에 그림 1-25 처럼 그림까지 떠요

개발환경 : macbook pro 16 2023 모델 (apple silicon, M2 Pro)
                mac os : Ventura 13.5
                iphone : iphone 13 mini
                ios: 16.6
                xcode: 14.3.1

 tsx 이면 typescript 파일인데 왜 js 자바 스크립트 파일이아닌 타입스크립트 파일이 생성된건가요?

리액트 네이티브 프로젝트 생성은
leedonghwan@mac-ldh ~ % npx react-native init LearnReactNative
                                                          
               ######                ######               
             ###     ####        ####     ###             
            ##          ###    ###          ##            
            ##             ####             ##            
            ##             ####             ##            
            ##           ##    ##           ##            
            ##         ###      ###         ##            
             ##  ########################  ##             
          ######    ###            ###    ######          
      ###     ##    ##              ##    ##     ###      
   ###         ## ###      ####      ### ##         ###   
  ##           ####      ########      ####           ##  
 ##             ###     ##########     ###             ## 
  ##           ####      ########      ####           ##  
   ###         ## ###      ####      ### ##         ###   
      ###     ##    ##              ##    ##     ###      
          ######    ###            ###    ######          
             ##  ########################  ##             
            ##         ###      ###         ##            
            ##           ##    ##           ##            
            ##             ####             ##            
            ##             ####             ##            
            ##          ###    ###          ##            
             ###     ####        ####     ###             
               ######                ######               
                                                          

                  Welcome to React Native!                
                 Learn once, write anywhere               

✔ Downloading template
✔ Copying template
✔ Processing template
✔ Installing Ruby Gems
✔ Installing CocoaPods dependencies (this may take a few minutes)

  
  Run instructions for Android:
    • Have an Android emulator running (quickest way to get started), or a device connected.
    • cd "/Users/leedonghwan/LearnReactNative" && npx react-native run-android
  
  Run instructions for iOS:
    • cd "/Users/leedonghwan/LearnReactNative" && npx react-native run-ios
    - or -
    • Open LearnReactNative/ios/LearnReactNative.xcworkspace in Xcode or run "xed -b ios"
    • Hit the Run button
    
  Run instructions for macOS:
    • See https://aka.ms/ReactNativeGuideMacOS for the latest up-to-date instructions.
    
  
leedonghwan@mac-ldh ~ % cd "/Users/leedonghwan/LearnReactNative" && npx react-native run-ios
info Found Xcode workspace "LearnReactNative.xcworkspace"
info Found booted lee의 iPhone
info Building (using "xcodebuild -workspace LearnReactNative.xcworkspace -configuration Debug -scheme LearnReactNative -destination id=00008110-001079420A50401E")

info 💡 Tip: Make sure that you have set up your development environment correctly, by running react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor 

error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening LearnReactNative.xcworkspace.
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace LearnReactNative.xcworkspace -configuration Debug -scheme LearnReactNative -destination id=00008110-001079420A50401E

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

Prepare packages

Computing target dependency graph and provisioning inputs

Create build description
Build description signature: 4c8197e1ad3dc548967ba9a5792bbbef
Build description path: /Users/leedonghwan/Library/Developer/Xcode/DerivedData/LearnReactNative-bpafbhbgdtmgwlcprotftgfmmkbt/Build/Intermediates.noindex/XCBuildData/4c8197e1ad3dc548967ba9a5792bbbef.xcbuilddata

note: Building targets in dependency order
/Users/leedonghwan/LearnReactNative/ios/LearnReactNative.xcodeproj: error: Signing for "LearnReactNative" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'LearnReactNative' from project 'LearnReactNative')
warning: Run script build phase 'Bundle React Native code and images' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'LearnReactNative' from project 'LearnReactNative')
warning: Run script build phase 'Start Packager' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'LearnReactNative' from project 'LearnReactNative')

2023-08-10 13:23:07.597 xcodebuild[22612:200856] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
** BUILD FAILED **

leedonghwan@mac-ldh LearnReactNative % 

2. 교재 61페이지
  App 컴포넌트 선언 부분 수정내용이 제 파일과 동일 하지 않아요
 변경전 내용인
 const App: ()=> React$Node = () => { 
 이 부분이
 제 파일에선
<img width="1728" alt="image" src="https://github.com/velopert/dealing-with-react-native/assets/7827525/9d70fdbb-e7e9-4326-81aa-0ad39d8adcbb">

처럼 다르게 표현되요,
물론,  처음부터 다른 파일 .tsx 라 여러가지 다들수 잇는데 
책 처음 부분부터 따라해보기를 하고 있는데

책 내용과 실행되는 내용이 다르다 보니
맥이 빠집니다. 출간된지 2년이 넘긴 햇지만...

제가 설정을 뭘 다르게 했는지 ...

그래서 이책의 오타 정정등
찾아봤는데 길벗, 님의 블로그, 님의 github를 찾아 봣지만 초창기에 올리신
코드만 있는듯요...

그리고 49페이지의 애플 실리콘 내용 부분도 저와는 다릅니다.
제가 사용중인 맥북이 애플 실리콘 에 해당하지만...

50페이지  yarn ios 하면 에러가 납니다.
대신 안내하는
  Run instructions for iOS:
    • cd "/Users/leedonghwan/LearnReactNative" && npx react-native run-ios
    - or -
    • Open LearnReactNative/ios/LearnReactNative.xcworkspace in Xcode or run "xed -b ios"
    • Hit the Run button

내용대로 하면
실물 iphone 대상으로 build를 하지만
에러가 나는데 sign 관련으로
xcode에서 team을 다른 거 할때 문제없이 되는 것으로 선택해주면
에러없이 진행이 됩니다....

책과 제가 하는것과 차이가 많지만 
그래도 책을 기준으로 따라 하다가...

61페이지 
App 컴포넌트 수정 부분부터 는 어떻게 해야될지...막혔습니다.

저는 타입스크립트에서 사용하는 tsx 구조 인듯하고
님은 javascript 기반의 js 파일인데....

**Comment by velopert** [comment]
```
npx react-native init LearnReactNative --version 0.70
```

0.71 부턴 기본 환경이 TypeScript 입니다. 따라서 위 명령어로 프로젝트를 생성하시면 됩니다. 
해당 내용 다음 인쇄부터 반영하겠습니다. 감사합니다.
